src/main/java/org/apache/maven/plugin/compiler/ModuleInfoTransformer.java

```diff
@@ -45,10 +53,11 @@ public ModuleVisitor visitModule(String name, int access, String version) {
                     @Override
                     public void visitRequire(String module, int access, String version) {
                         // Check if the module name matches the java/jdk modules
-                        if (module.startsWith("java.") || module.startsWith("jdk.")) {
+                        if (version != null && (module.startsWith("java.") || module.startsWith("jdk."))) {
```

Comment:
This null check will preserve the current behavior when compiling from a newer JDK to an older target (eg. using JDK 11 to compile the module-info using --release 9).
In other words, it will not patch anything, while this is perfectly valid, the JDK-8318913 aims to modify this behavior to set the simple version **always**, I'm ok to preserve the current behavior and check later if we need to drop or not this null check.